app/controllers/api/v1/merchants/search_controller.rb

```diff
@@ -0,0 +1,15 @@
+class Api::V1::Merchants::SearchController < ApplicationController
+
+  def show
+    render json: Merchant.find_by(name: params[:name]) if params[:name]
```

Comment:
I like that you have this working. Let's refactor so that we are pulling the params with a name like "search_params" and finds whatever we're looking for. :)